eventstore/storethread.go

```diff
@@ -120,26 +135,26 @@ func (a *singleThreadAdapter) threadToStore(wg *sync.WaitGroup) {
 }
 
 func (a *singleThreadAdapter) storeToThread(wg *sync.WaitGroup) {
+	defer a.goRoutines.Done()
 	l := a.store.localEventListen()
 	defer l.Discard()
 	wg.Done()
 
 	for {
 		select {
-		case <-a.ctx.Done():
-			log.Infof("cancelling sending store local events to own thread log for thread %s", a.threadID)
+		case <-a.closeChan:
+			log.Infof("closing store-to-thread flow on thread %s", a.threadID)
 			return
 		case event, ok := <-l.Channel():
 			if !ok {
-				log.Infof("ending sending store local event to own thread since channel was closed for thread %s", a.threadID)
+				log.Errorf("ending sending store local event to own thread since channel was closed for thread %s", a.threadID)
```

Comment:
Same as above comment.
For some reason the Store closed our statechanged listener... switched from `Info` to `Error` since shouldn't be OK.